File: backend/src/main/java/com/bookapp/model/Review.java
package com.bookapp.model;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "reviews")
public class Review {

    @Id
    private String id;

    private String userId;
    private String userName;
    private String bookId;

    private int rating; // 1-5
    private String review;

    private LocalDateTime createdAt;
}

File: BookApp/app/src/main/java/com/bookapp/data/model/Review.kt
package com.bookapp.data.model

data class Review(
    val id: String? = null,
    val userId: String? = null,
    val username: String? = null,
    val fullName: String? = null,
    val avatar: String? = null,
    val bookId: String? = null,
    val rating: Int? = null,
    val review: String? = null,
    val createdAt: String? = null
)
